packages/redux_persist_flutter/lib/redux_persist_flutter.dart

```diff
@@ -95,9 +112,62 @@ class SharedPreferencesEngine implements StorageEngine {
   @override
   Future<void> save(Uint8List? data) async {
     final sharedPreferences = await _getSharedPreferences();
-    sharedPreferences.setString(key, uint8ListToString(data)!);
+    sharedPreferences.setString(key, await uint8ListToString(data) ?? '');
   }
 
   Future<SharedPreferences> _getSharedPreferences() async =>
       await SharedPreferences.getInstance();
 }
+
+class FlutterJsonSerializer<T> implements StateSerializer<T> {
+  /// Turns the dynamic [json] (can be null) to [T]
+  final T? Function(dynamic? json) decoder;
+
+  FlutterJsonSerializer(this.decoder);
+
+  @override
+  Future<T?> decode(Uint8List? data) async {
+    return decoder(
+        data != null ? json.decode(await uint8ListToString(data) ?? '') : null);
```

Comment:
Keep it consistent with what they've already got. Do this instead:
`json.decode((await uint8ListToString(data))!)`